File: primary-algorithm/src/main/java/moon/scala/TwoSum.scala
package moon.scala

class TwoSum {
  def twoSum(nums:Array[Int],target:Int):Array[Int]= {
    val list=nums.toList
    var a=0
    def loop(head:Int,tail:List[Int]):Array[Int]={
      if(tail.contains(target-head)){
        Array(a,a+tail.indexOf(target-head)+1)
      }else{
        a+=1
        if(tail.tail!=Nil)
        loop(tail.head,tail.tail)
        else
          throw new IllegalArgumentException("数组不满足条件")
      }
    }
    loop(list.head,list.tail)
  }
}

File: data-structure/src/main/java/array/TwoSum.java
package array;

public class TwoSum {
    public int[] solve(int[] numbers, int target) {
        int i=0,j=numbers.length-1;
        while(i<j){
            if(numbers[i]+numbers[j]>target){
                j--;
                continue;
            }
            if(numbers[i]+numbers[j]<target){
                i++;
                continue;
            }
            return new int[]{++i,++j};
        }
        return null;
    }
}
